### PR TITLE
system-loadsave.c: Protect state file from world-readable

### PR DESCRIPTION
**Pull request by zdohnal**
The state file might include user-sensitive data, we should prevent accessing it from others than owner.

### CODE REVIEW

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
This usage isn't safe. You need to use fchmod right after the file is opened, or wrap the open with umask calls. You also need to not error out if the chmod fails since some filesystems/platforms don't support the operation with all values (think Windows in particular...)

Look at what cupsd does in scheduler/file.c with config/state files...

**Comment by zdohnal** [comment]
@michaelrsweet thx, I'll look into it!

**Comment by zdohnal** [comment]
I have found more files which can be protected, so I've added them as well.

File: pappl/system-webif.c [hunk 1]
```diff
@@ -3308,6 +3324,9 @@ tls_make_certsignreq(
   }
   else if ((fp = cupsFileOpen(keyfile, "w")) != NULL)
   {
+    if (fchmod(cupsFileNumber(fp), 0600))
```
Comment (michaelrsweet) [inline pappl/system-webif.c:3327]: A signing request isn't security-critical...

**Comment by michaelrsweet** [comment]
[master bf52117] Specify restricted permissions for the state file.

The cert/key/csr code is getting overhauled to use the new CUPS X.509 APIs... (#366)

**Pull request closed without merging**